v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go

```diff
@@ -409,26 +409,33 @@ func getOwnerFunction(
 	}
 
 	owner := astbuilder.Selector(specSelector, astmodel.OwnerProperty)
+	nilOwnerCheck := astbuilder.IfNil(owner, astbuilder.Returns(astbuilder.Nil()))
+	nilOwnerCheck.Decs.After = dst.EmptyLine
 
 	switch r.Resource().Scope() {
 	case astmodel.ResourceScopeResourceGroup:
 		fn.AddComments("returns the ResourceReference of the owner")
 		fn.AddStatements(
+			nilOwnerCheck,
```

Comment:
Are these checks just being defensive? I'm not 100% following them (though I don't mind them if it's just being defensive either).

Reply:
Defensive checks added as a result of new unit tests that discovered an issue in `admissions.go`.